.github/workflows/CICD.yml

```diff
@@ -842,6 +850,10 @@ jobs:
       with:
         persist-credentials: false
     - uses: Swatinem/rust-cache@v2
+    - name: Install/setup prerequisites
```

Comment:
I would avoid to have two steps with the same name in the same job. So my suggestion is to either rename this step or the one on line 859.